### PR TITLE
selectionChange fires when input is emptied

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Issue: https://github.com/blackbaud/skyux-lookup/issues/22

### CODE REVIEW

File: src/app/public/modules/autocomplete/autocomplete.component.ts [hunk 1]
```diff
@@ -148,6 +148,7 @@ export class SkyAutocompleteComponent
   private isMouseEnter = false;
   private searchResultsIndex = 0;
   private searchText: string;
+  private previousValue: string;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/autocomplete/autocomplete.component.ts]: Storing the `ControlValueAccessor`'s previous value within the wrapping component feels off to me. The autocomplete component (in theory) should only be dealing with the search text and the dropdown. The value is handled by the `SkyAutocompleteDirective`. Any way we can isolate value checks to the directive?

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/autocomplete/autocomplete.component.ts]: I also noticed that the input directive watches for blur events, if that's useful:
https://github.com/blackbaud/skyux-lookup/blob/master/src/app/public/modules/autocomplete/autocomplete-input.directive.ts#L89-L94

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/autocomplete/autocomplete.component.ts]: Hmm, valid concern. I started here, but then realized the `SkyAutocompleteComponent` only emits the event when a search selection is made. No matter what the direction we go in, we'll need this component to be aware of blur and emit an event if the text is cleared. That's why I landed on this solution. Without storing the previous version, the emitter would fire on every blur, regardless if the value had changed.

Ideas on a different approach?

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/autocomplete/autocomplete.component.ts]: We could add a new output event to the directive, like `reset` or `cleared`...

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/autocomplete/autocomplete.component.ts]: I approached this a different way. What do you think?

**Pull request merged**